src/test/BadRouter.sol

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.24;
+
+import {CurrencyLibrary, Currency} from "../types/Currency.sol";
+import {IPoolManager} from "../interfaces/IPoolManager.sol";
+import {BalanceDelta, BalanceDeltaLibrary} from "../types/BalanceDelta.sol";
+import {PoolKey} from "../types/PoolKey.sol";
+import {IHooks} from "../interfaces/IHooks.sol";
+import {Hooks} from "../libraries/Hooks.sol";
+import {PoolTestBase} from "./PoolTestBase.sol";
+import {Hooks} from "../libraries/Hooks.sol";
+import {IHooks} from "../interfaces/IHooks.sol";
+import {IERC20Minimal} from "../interfaces/external/IERC20Minimal.sol";
+
+contract BadRouter is PoolTestBase {
```

Comment:
LOL it took me a second to figure out why the router is bad. Could you add a comment above just saying that its bc it doesnt call sync?